### PR TITLE
order post request completed

### PR DESCRIPTION
**Pull request by jinho6225**
Screen Shot 2020-03-10 at 5 32 19 PM

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -116,6 +116,38 @@ app.post('/api/cart', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+  if (!req.session.cartId) {
+    res.status(400).json({
```
Comment (thebearingedge) [inline server/index.js:121]: Be sure to `return` to prevent more code from executing.

File: server/index.js [hunk 2]
```diff
@@ -116,6 +116,38 @@ app.post('/api/cart', (req, res, next) => {
...
+      error: 'there is not cardId'
+    });
+  }
+  const { name, creditCard, shippingAddress } = req.body;
+  if (name === undefined || creditCard === undefined || shippingAddress === undefined || typeof creditCard !== 'number') {
+    res.status(400).json({
```
Comment (thebearingedge) [inline server/index.js:128]: Be careful to `return` so that your server does not try to respond twice and crash.

File: server/index.js [hunk 3]
```diff
@@ -116,6 +116,38 @@ app.post('/api/cart', (req, res, next) => {
...
+      `;
+  const params = [cartId, name, creditCard, shippingAddress];
+  db.query(sql, params)
+    .then(result => {
+      const order = {};
+      order.createdAt = result.rows[0].createdAt;
+      order.creditCard = result.rows[0].creditCard;
+      order.name = result.rows[0].name;
+      order.orderId = result.rows[0].orderId;
+      order.shippingAddress = result.rows[0].shippingAddress;
```
Comment (thebearingedge) [inline server/index.js]: The `result.rows[0]` is what you wanted.
```js
res.status(201).json(result.rows[0])
```

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 3 above)
Comment (jinho6225) [reply server/index.js]: instruction response example dose not have cartId;
so i created newObj, but okay! changed it

File: server/index.js (hunk 2 above)
Comment (jinho6225) [reply server/index.js:128]: got it!

File: server/index.js (hunk 1 above)
Comment (jinho6225) [reply server/index.js:121]: got it!! changed it!

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**